bpf/init.sh

```diff
@@ -141,17 +142,20 @@ function setup_proxy_rules()
 		ip route replace table $PROXY_RT_TABLE default via $IP4_HOST
 	fi
 
-	if [ -n "$(ip -6 rule list)" ]; then
-		if [ -z "$(ip -6 rule list $rulespec)" ]; then
-			ip -6 rule add $rulespec
-		fi
-	fi
-
-	IP6_LLADDR=$(ip -6 addr show dev $HOST_DEV2 | grep inet6 | head -1 | awk '{print $2}' | awk -F'/' '{print $1}')
-	if [ -n "$IP6_LLADDR" ]; then
-		ip -6 route replace table $PROXY_RT_TABLE ${IP6_LLADDR}/128 dev $HOST_DEV1
-		ip -6 route replace table $PROXY_RT_TABLE default via $IP6_LLADDR dev $HOST_DEV1
-	fi
+    # PRE_EXISTING_DEVICE might not have an IPv6 address
+    if [ -z "${PRE_EXISTING_DEVICE}" ]; then
```

Comment:
I was assuming that the CNI could potentially support IPv6 in future.

The existing code checks whether the `$IP6_HOST` variable is set and configures IPv6 depending on that. This new version assumes that if the policy enforcement interface is set, then IPv6 is not supported. If this is the case, nothing is logged.

I would have expected that in the daemon code, we figure out whether IPv6 support is enabled or not and pass down `$IP6_HOST` as `<nil>` in the flannel case, which shouldn't require this kind of change to the `init.sh` script. Then in the logic where we make this decision in the daemon, we could log at info/debug level saying something like "policy enforcement interface is set, disabling IPv6".

Not a big deal since the current CNI we integrate with doesn't support IPv6, I was just thinking that at some point if we want to integrate with a CNI that does support this, then this hunk will need to be reverted and the logic for whether to configure IPv6 moved into the daemon.